compiler/rustc_resolve/src/macros.rs

```diff
@@ -500,6 +502,18 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             Err(Determinacy::Undetermined) => return Err(Indeterminate),
         };
 
+        if kind == MacroKind::Attr
```

Comment:
```suggestion
        if res == Res::NonMacroAttr(NonMacroAttrKind::Tool)
```
This is incorrect because we can encounter macro attribute resolutions here.

This logic should also be run at the very end of `smart_resolve_macro_path`, after all the "unexpected resolution" errors are reported.

Reply:
The second part of the comment wasn't addressed.